exports/defect_dojo.py

```diff
@@ -0,0 +1,138 @@
+                    f"Defect Dojo did not answer as expected during login (status: {resp.getcode()})"
+                )
+
+            self.token = json.load(resp)["token"]
+
+        self.headers["Authorization"] = f"Token {self.token}"
+        logging.debug("Defect Dojo: successfully refreshed token")
+
+    def engagement_exists(self, engagement_id=None, name=None):
+        """Return True if an engagement exists, False otherwise
+        Engagement is identified either by its name or its ID (positive integer)"""
+        if not self.token:
+            self.get_token()
+        if engagement_id:
+            resp = requests.get(
+                f"{self.base_url}/api/v2/engagements/?engagment={engagement_id}",
+                headers=self.headers,
+            )
+        elif name:
+            resp = requests.get(
+                f"{self.base_url}/api/v2/engagements/?name={parse.quote_plus(name)}",
+                headers=self.headers,
+            )
+        else:
+            raise ValueError("Either an engagement name or ID must be provided")
+
+        if resp.status_code >= 400:
+            logging.warning(
+                f"Error while looking for engagement ({resp.status_code}, {resp.get('message')})"
+            )
+        counts = resp.json()["counts"]
+        if counts > 1:
+            logging.warning("Error while looking for engagement: too many hits")
+        return counts >= 1
+
+    def _private_import(self, endpoint, data, filename):
+        """Send a POST request to endpoint [typically import or reimport], with data and filname"""
+        if not self.token:
+            self.get_token()
+
+        resp = requests.post(
+            endpoint,
+            headers=self.headers,
+            data=data,
+            files={"file": open(filename, "rb")},
+        )
+        if resp.status_code >= 400:
+            print(vars(resp))
+            err = resp.json()
+            logging.warning(f"Error while exporting ({resp.status_code}, {err})")
+
+    def reimport_scan(self, data, filename):
+        """Reimport to an existing engagement with an existing compatible scan."""
+
+        mandatory = {
+            "product_name",
+            "engagement_name",
+            "scan_type",
+            "active",
+            "verified",
+        }
+        missing = mandatory - set(data.keys())
+        if missing:
+            raise ValueError(f"Missing required entries for reimport: {missing}")
+
+        self._private_import(f"{self.base_url}/api/v2/reimport-scan/", data, filename)
+
+    def import_scan(self, data, filename):
+        """Import to an existing engagement."""
+
+        mandatory = {
+            "product_name",
+            "engagement_name",
+            "scan_type",
+            "active",
+            "verified",
+        }
+        missing = mandatory - set(data.keys())
+        if missing:
+            raise ValueError(f"Missing required entries for reimport: {missing}")
```

Comment:
the lines 111-122 are duplicated. how about moving it into _private_import or in a new method?

Reply:
oh... that's something I forgot. Improvement on the way!

Reply:
improved: 
- each import/reimport function do verify that necessary data is there for their own endpoint
- the _private method checks that the other 3 generic parameters present